### PR TITLE
Modify clob handling

### LINKED ISSUE #6078: loadData for csv file with name column
### Search first

- [X] I searched and no similar issues were found

### Description

If you want to use Liquibase loadData for a CSV file and there is a column 'name' you get the following error.
Liquibase is searching for a file 'Afghanistan' which is content of the CSV file.
If you import a CSV file without a column 'name' everything is working.

ERROR: Exception Primary Class:  FileNotFoundException
ERROR: Exception Primary Reason:  The file Afghanistan was not found in the configured search path:
...
2024-07-06 14:50:09.957 SEVERE liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase.xml::2024-07-06_0001::stegra:
     Reason: liquibase.exception.DatabaseException: The file Afghanistan was not found in the configured search path:
...
liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase.xml::2024-07-06_0001::stegra:
...
More locations can be added with the 'searchPath' parameter.
	at liquibase.statement.ExecutablePreparedStatementBase.applyColumnParameter(ExecutablePreparedStatementBase.java:345)
	at liquibase.statement.ExecutablePreparedStatementBase.attachParams(ExecutablePreparedStatementBase.java:217)
	at liquibase.statement.ExecutablePreparedStatementBase.execute(ExecutablePreparedStatementBase.java:117)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:175)
	at liquibase.executor.AbstractExecutor.execute(AbstractExecutor.java:141)
	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1176)
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:768)
...
Caused by: java.io.FileNotFoundException: The file Afghanistan was not found in the configured search path:
...
More locations can be added with the 'searchPath' parameter.
	at liquibase.resource.ResourceAccessor.getExisting(ResourceAccessor.java:222)
	at liquibase.statement.ExecutablePreparedStatementBase.getResourceAsStream(ExecutablePreparedStatementBase.java:479)
	at liquibase.statement.ExecutablePreparedStatementBase.toCharacterStream(ExecutablePreparedStatementBase.java:435)
	at liquibase.statement.ExecutablePreparedStatementBase.applyColumnParameter(ExecutablePreparedStatementBase.java:338)

### Steps To Reproduce

Content of data/countries_iso_3166.csv:
id,name,alpha_2,alpha_3
4,Afghanistan,AF,AFG

Liquibase changeSet:
```
<changeSet author="stegra" id="2024-07-06_0001">
<loadData encoding="UTF-8" relativeToChangelogFile="true" file="data/countries_iso_3166.csv"
	separator="," schemaName="public" tableName="country">
</loadData>
</changeSet>
```

Java code:
Scope.child(
		Map.of(Scope.Attr.resourceAccessor.name(),
				new ClassLoaderResourceAccessor(getClass().getClassLoader())),
		() -> new CommandScope("update")
				.addArgumentValue(DbUrlConnectionArgumentsCommandStep.DATABASE_ARG, database)
				.addArgumentValue(UpdateCommandStep.CHANGELOG_FILE_ARG, LIQUIBASE_CHANGELOG)
				.addArgumentValue(DatabaseChangelogCommandStep.CHANGELOG_PARAMETERS,
						new ChangeLogParameters(database))
				.execute());

### Expected/Desired Behavior

A 'name' column should be treated as every other column.

### Liquibase Version
4.28.0

### Database Vendor & Version

PostgreSQL 16.3.0

### Liquibase Integration

Java

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

Ubuntu 24.04

### Additional Context

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

### PR DESCRIPTION
**Pull request by tati-qalified**
## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 

## Description
Fixes #6078 and #6068
For the `loadData` change, currently all `CLOB` type columns are expected to be a path to a file. This isn't the intuitive behaviour, because `text` and `longvarchar` are `CLOB` types. 
This PR checks the string passed to a `CLOB` type column. By default it will be interpreted as a path. If the file at that path exists, the file's contents will be inserted into the column. If not, the passed value will be inserted. 

## Things to be aware of
If a user is intending to point to a file but it doesn't exist, instead of exiting with an error, the string will be inserted into the column. **This is where documentation is needed.**

### CODE REVIEW

**Comment by wwillard7800** [comment]
I think you may need to do a merge from master for this PR.

**Comment by filipelautert** [comment]
> I think you may need to do a merge from master for this PR.

@wwillard7800 fixed, thanks

**Review by filipelautert — APPROVED** [review]
Nice work @tati-qalified !

**Comment by filipelautert** [comment]
@rberezen could you QA this one as it was developed by @tati-qalified ?

File: liquibase-standard/SQLFileTestData.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+TESTDATA
```
Comment (rberezen) [inline liquibase-standard/SQLFileTestData.sql:1]: is it a right place to store test-files? @tati-qalified @MalloD12 @filipelautert

Comment (filipelautert) [reply liquibase-standard/SQLFileTestData.sql:1]: Whops, this file should have been inside liquibase-standard/src/test/resources/liquibase/change/core/ directory (and it was afaik).  Could you fix it @tati-qalified ?

**Comment by tati-qalified** [comment]
@filipelautert I've been looking into the Oracle test that fails, and I think that it's because the path inside the csv file is relative to that csv file, and not relative to the package from where the test is executed.

In my test `liquibase-standard/src/test/groovy/liquibase/change/core/LoadDataChangeTest.groovy` I'm using a csv file with a path relative to `test/resources`.

What do you suggest I do?

**Comment by filipelautert** [comment]
> In my test `liquibase-standard/src/test/groovy/liquibase/change/core/LoadDataChangeTest.groovy` I'm using a csv file with a path relative to `test/resources`.
> 
> What do you suggest I do?

@tati-qalified I believe that moving the file that is in root directory to  liquibase-standard/src/test/resources/liquibase/change/core/ directory could fix this. Can we try it?

**Review by rberezen — APPROVED** [review]
@tati-qalified great job!

**Pull request merged**